iotdb/src/main/java/org/apache/iotdb/db/query/executor/EngineExecutorWithoutTimeGenerator.java

```diff
@@ -110,25 +127,43 @@ public QueryDataSet executeWithoutFilter()
       QueryDataSource queryDataSource = QueryDataSourceManager.getQueryDataSource(jobId, path);
 
       // add data type
-      dataTypes.add(MManager.getInstance().getSeriesType(path.getFullPath()));
+      try {
+        dataTypes.add(MManager.getInstance().getSeriesType(path.getFullPath()));
+      } catch (PathErrorException e) {
+        throw new FileNodeManagerException(e);
+      }
 
       PriorityMergeReader priorityReader = new PriorityMergeReader();
 
       // sequence insert data
-      SequenceDataReader tsFilesReader = new SequenceDataReader(queryDataSource.getSeqDataSource(),
-          null);
-      priorityReader.addReaderWithPriority(tsFilesReader, 1);
+      SequenceDataReader tsFilesReader = null;
+      try {
+        tsFilesReader = new SequenceDataReader(queryDataSource.getSeqDataSource(),
+            null);
+        priorityReader.addReaderWithPriority(tsFilesReader, 1);
+      } catch (IOException e) {
+        throw new FileNodeManagerException(e);
+      }
 
       // unseq insert data
-      PriorityMergeReader unSeqMergeReader = SeriesReaderFactory.getInstance()
-          .createUnSeqMergeReader(queryDataSource.getOverflowSeriesDataSource(), null);
-      priorityReader.addReaderWithPriority(unSeqMergeReader, 2);
+      PriorityMergeReader unSeqMergeReader = null;
+      try {
+        unSeqMergeReader = SeriesReaderFactory.getInstance()
+            .createUnSeqMergeReader(queryDataSource.getOverflowSeriesDataSource(), null);
+        priorityReader.addReaderWithPriority(unSeqMergeReader, 2);
+      } catch (IOException e) {
+        throw new FileNodeManagerException(e);
+      }
 
       readersOfSelectedSeries.add(priorityReader);
     }
 
-    return new EngineDataSetWithoutTimeGenerator(queryExpression.getSelectedSeries(), dataTypes,
-        readersOfSelectedSeries);
+    try {
+      return new EngineDataSetWithoutTimeGenerator(queryExpression.getSelectedSeries(), dataTypes,
```

Comment:
Using `FileNodeManagerException` to encapsulate the `IOException` and `PathErrorException` is confusing.

Reply:
discussed

Reply:
Yeah, FileNodeManagerException is a high-layer exception. If possible, it should not be the direct exception to make problems more trackable or classifiable.

Reply:
Agreed. I would recommend a thorough investigation into our exceptions in the future.